packages/@webex/plugin-meetings/src/multistream/sendSlotManager.ts

```diff
@@ -0,0 +1,162 @@
+import {
+  SendSlot,
+  MediaType,
+  LocalStream,
+  MultistreamRoapMediaConnection,
+} from '@webex/internal-media-core';
+
+export default class SendSlotManager {
+  private readonly slots: Map<MediaType, SendSlot> = new Map();
+  private readonly LoggerProxy: any;
+
+  constructor(LoggerProxy: any) {
+    this.LoggerProxy = LoggerProxy;
+  }
+
+  /**
+   * This method is used to create a sendSlot for the given mediaType and returns the created sendSlot
+   * @param {MultistreamRoapMediaConnection} mediaConnection MultistreamRoapMediaConnection for which a sendSlot needs to be created
+   * @param {MediaType} mediaType MediaType for which a sendSlot needs to be created (AUDIO_MAIN/VIDEO_MAIN/AUDIO_SLIDES/VIDEO_SLIDES)
+   * @param {boolean} active This is optional boolean to set the active state of the sendSlot. Default is true
+   * @returns {SendSlot} slot The created sendSlot
+   */
+  public createSlot(
+    mediaConnection: MultistreamRoapMediaConnection,
+    mediaType: MediaType,
+    active = true
+  ): SendSlot {
+    if (this.slots.has(mediaType)) {
+      throw new Error(`Slot for ${mediaType} already exists`);
+    }
+
+    const slot: SendSlot = mediaConnection.createSendSlot(mediaType, active);
+
+    this.slots.set(mediaType, slot);
+
+    this.LoggerProxy.logger.info(
+      `SendSlotsManager->createSlot#Created slot for ${mediaType} with active ${active}`
+    );
+
+    return slot;
+  }
+
+  /**
+   * This method is used to retrieve the sendSlot for the given mediaType
+   * @param {MediaType} mediaType of which the slot needs to be retrieved
+   * @returns {SendSlot}
+   */
+  public getSlot(mediaType: MediaType): SendSlot {
```

Comment:
this is not called from anywhere, so maybe not needed?

Reply:
We use this in tests now extensively as we do not stub the sendSlotManager, but rather the sendSlot itself. 

That said, we decided to have it there just in case in the future if we need to check if a slot is active or not, we shall getSlot and then check its `active` property. 

If this really needs to go away, we can make it go away with the `isActive(MediaType)` method but I'd like to take that as a TODO. We will have to fix tests as well if we are removing this now.

Reply:
ok, that's fine we can keep it